tests/neg/i4659b.scala

```diff
@@ -1,5 +0,0 @@
-case class SourcePosition(outer: SourcePosition = NoSourcePosition) {
-  assert(outer != null)  // crash
```

Comment:
`tests/run/i4659b.scala` covers the original issue. This particular error is not emitted anymore. The actual issue should be caught by the initialization checker.

Reply:
@liufengyun I tried `dotc tests/run/i4659b.scala -Ycheck-init` but it did not seem to catch it. Is this expected?

Reply:
Yes, the checker does not check global/static fields, because that would require global analysis.